disnake/client.py

```diff
@@ -271,22 +271,37 @@ def __init__(
         *,
         asyncio_debug: bool = False,
         loop: Optional[asyncio.AbstractEventLoop] = None,
-        **options: Any,
+        shard_id: Optional[int] = None,
+        shard_count: Optional[int] = None,
+        enable_debug_events: bool = False,
+        connector: Optional[aiohttp.BaseConnector] = None,
+        proxy: Optional[str] = None,
+        proxy_auth: Optional[aiohttp.BasicAuth] = None,
+        assume_unsync_clock: bool = True,
+        max_messages: Optional[int] = 1000,
+        application_id: Optional[int] = None,
+        heartbeat_timeout: float = 60.0,
+        guild_ready_timeout: float = 2.0,
+        allowed_mentions: Optional[AllowedMentions] = None,
+        activity: Optional[BaseActivity] = None,
+        status: Optional[Union[Status, str]] = None,
+        intents: Intents = Intents.default(),
+        chunk_guilds_at_startup: Optional[bool] = None,
```

Comment:
This won't work due to
https://github.com/DisnakeDev/disnake/blob/b40a268a6bed609874c39a8782159bf980fa2529/disnake/state.py#L237
Since you can't really set a proper default value for the parameter here, `ConnectionState.__init__` should be updated to handle a `None` (or `utils.MISSING`?) value

Reply:
Fixed by replacing with
```py
chunk_guilds = options.get("chunk_guilds_at_startup")
self._chunk_guilds: bool = intents.members if chunk_guilds is None else chunk_guilds
```
It should now be set to default even if `None` was passed.